src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -84,6 +84,12 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
   struct SubmittedMarker {
     SubmissionMetaInformation meta_information;
     uint32_t slot_index;
+
+    // These are set when the timestamps were queried successfully. When these have values, the
+    // corresponding slot indices for timestamp queries have been reset. Do not query again in
+    // that case and only get the timestamps through these members.
+    std::optional<uint64_t> begin_timestamp;
```

Comment:
I guess with the new change, you only need one timestamp here.
A slice `SubmittedMarker`s for both start and end (well start is optional...). So in the code below you are always accessing either end_marker.end_timestamp or begin_marker->begin_timestamp.

Reply:
Ah, you are right, will fix right away.

Reply:
Done.